src/lib/i18next.js

```diff
@@ -0,0 +1,60 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2021 Graz University of Technology.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+// I18next dependencies
+// npm install react-i18next i18next i18next-browser-languagedetector --save
+
+import i18next from 'i18next';
+
+// import translations js
+// import { TRANSLATIONS_DE } from './translations/de/translations';
+// import { TRANSLATIONS_EN } from './translations/en/translations';
+
+// import translations json
+// (Note that you need @rollup/plugin-json to import JSON files)
+// rollup.config.js
+//  - import json from "@rollup/plugin-json";
+//  - json(),
+import TRANSLATIONS_DE from './translations/de/translations.json';
+import TRANSLATIONS_EN from './translations/en/translations.json';
+
+
+// https://github.com/i18next/i18next-browser-languageDetector
+import LanguageDetector from 'i18next-browser-languagedetector';
+import { initReactI18next } from 'react-i18next';
+
+i18next
+  // detect user language
+  // learn more: https://github.com/i18next/i18next-browser-languageDetector
+  .use(LanguageDetector)
+  // pass the i18n instance to react-i18next.
+  .use(initReactI18next)
+  // init i18next
+  // for all options read: https://www.i18next.com/overview/configuration-options
+  .init({
+    fallbackLng: 'en',
+    debug: true,
+    resources: {
+      en: {
```

Comment:
I assume we would have to embed the translations of all languages right? I guess it's not too big an overhead, compared to having to find a way to only include the once an instance are using.

Reply:
i18next - supports different approaches to add the resources(translations), see also [load-translations](https://www.i18next.com/how-to/add-or-load-translations), in this example I used something similar to python modules.